File: java/template-renderer/src/main/java/com/rbleggi/templaterenderer/core/RendererFactory.java
package com.rbleggi.templaterenderer.core;

import com.rbleggi.templaterenderer.renderers.CSVRenderer;
import com.rbleggi.templaterenderer.renderers.HTMLRenderer;
import com.rbleggi.templaterenderer.renderers.PDFRenderer;

public class RendererFactory {
    public static TemplateRenderer getRenderer(String type) {
        return switch (type.toLowerCase()) {
            case "html" -> new HTMLRenderer();
            case "csv" -> new CSVRenderer();
            case "pdf" -> new PDFRenderer();
            default -> throw new IllegalArgumentException("Unknown format: " + type);
        };
    }
}

File: kotlin/template-renderer/src/main/kotlin/com/rbleggi/templaterenderer/core/RendererFactory.kt
package com.rbleggi.templaterenderer.core

import com.rbleggi.templaterenderer.renderers.CSVRenderer
import com.rbleggi.templaterenderer.renderers.HTMLRenderer
import com.rbleggi.templaterenderer.renderers.PDFRenderer

object RendererFactory {
    fun getRenderer(type: String): TemplateRenderer {
        return when (type.lowercase()) {
            "html" -> HTMLRenderer()
            "csv" -> CSVRenderer()
            "pdf" -> PDFRenderer()
            else -> throw IllegalArgumentException("Unknown format: $type")
        }
    }
}